pyro/poutine/posterior_messenger.py

```diff
@@ -0,0 +1,11 @@
+from .replay_messenger import ReplayMessenger
+
+
+class SamplePosteriorMessenger(ReplayMessenger):
+    # This acts like ReplayMessenger but additionally replays cond_indep_stack.
+
+    def _pyro_sample(self, msg):
+        if msg["infer"].get("enumerate") == "parallel":
+            super(SamplePosteriorMessenger, self)._pyro_sample(msg)
+        if msg["name"] in self.trace:
```

Comment:
:+1: this is a nice trick. @martinjankowiak @eb8680 do you think we could use something like this to support subsampling in autoguides?

Reply:
Thanks! Unfortunately, this is something I can not take credit for.
It was just a class I moved from `infer.discrete`, because I found it useful 😄

Reply:
Git history shows it is probably @eb8680 who implemented it.